ext/dom/php_dom.c

```diff
@@ -1302,6 +1302,10 @@ int dom_hierarchy(xmlNodePtr parent, xmlNodePtr child)
 {
 	xmlNodePtr nodep;
 
+	if (child->type == XML_DOCUMENT_NODE) {
```

Comment:
line 1309 tests for `child == NULL`, this check should be below that.

Reply:
Oops! Thank you!